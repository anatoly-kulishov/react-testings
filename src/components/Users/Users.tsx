import React, { useEffect, useState } from 'react';
import axios from 'axios';

const Users = () => {
  const [users, setUsers] = useState<any>([]);
  
  const loadUsers = async () => {
    const res = await axios.get('https://jsonplaceholder.typicode.com/users');
    setUsers(res.data);
  }
  
  useEffect(() => {
    loadUsers();
  }, [])
  
  return (
    <div>
      {users.map((user: any) => <div key={user.id} data-testid="user-item">{user.name}</div>)}
    </div>
  );
};

export default Users;
