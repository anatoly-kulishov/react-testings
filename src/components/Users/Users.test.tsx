import React from 'react';
import { render, screen } from '@testing-library/react';
import axios from "axios";
import Users from "./Users";

jest.mock('axios');
const mockedAxios = axios as jest.Mocked<typeof axios>;

describe('Users test', () => {
  let response: any;
  
  beforeEach(() => {
    response = {
      data: [
        {
          "id": 1,
          "name": "Leanne Graham",
        },
        {
          "id": 2,
          "name": "Ervin Howell",
        },
        {
          "id": 3,
          "name": "Clementine Bauch",
        }
      ]
    }
  })
  
  test('1', async () => {
    mockedAxios.get.mockReturnValue(response);
    render(<Users/>);
    const users = await screen.findAllByTestId('user-item');
    expect(users.length).toBe(3);
    expect(users).toMatchSnapshot();
  });
  
})

