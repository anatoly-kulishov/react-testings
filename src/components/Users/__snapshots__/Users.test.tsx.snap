// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST Users 1 1`] = `
Array [
  <div
    data-testid="user-item"
  >
    Leanne Graham
  </div>,
  <div
    data-testid="user-item"
  >
    Ervin Howell
  </div>,
  <div
    data-testid="user-item"
  >
    Clementine Bauch
  </div>,
]
`;

exports[`Users test 1 1`] = `
Array [
  <div
    data-testid="user-item"
  >
    Leanne Graham
  </div>,
  <div
    data-testid="user-item"
  >
    Ervin Howell
  </div>,
  <div
    data-testid="user-item"
  >
    Clementine Bauch
  </div>,
]
`;
